### PR TITLE
DM-25978: Add an example measurement and aggregator for extras

### PR DESCRIPTION
**Pull request by SimonKrughoff**

### CODE REVIEW

**Comment by SimonKrughoff** [comment]
To be clear, this was intended purely as a demonstration of functionality, not necessarily to produce something broadly useful.  If there is an obvious thing to do with that histogram, I'm happy to change the default.  I just thought the mode was an easy to compute and easy to understand quantity.

To answer 1) I think the patch level extras should definitely be packaged with the extras.  At the aggregated level, I don't think they should, since the aggregation is either over the metric measurement (AM1) or the extras.

File: python/metric_pipeline_tasks/MatchedCatalogMeasureTasks.py [hunk 1]
```diff
@@ -204,6 +204,51 @@ def run(self, matchedCatalog, metric_name):
...
+    return [i*delta for i in range(n+1)]
+
+
+class AMxWithHistTaskConfig(AMxTaskConfig):
+    bins = ListField(doc="Bins for histogram.",
+                     dtype=float, minLength=2, maxLength=1500,
+                     listCheck=isSorted, default=bins(30, 200))
+
+
+class AMxWithHistTask(Task):
```
Comment (bechtol) [inline python/metric_pipeline_tasks/MatchedCatalogMeasureTasks.py:222]: This run method is almost identical to the run method in the AMxTask, with the difference being the addition of the `extras` here. If this method is intended mainly as a demonstration, I think this is OK, but we might want to consolidate in the future if we begin using the `extras` more regularly.

Comment (SimonKrughoff) [reply python/metric_pipeline_tasks/MatchedCatalogMeasureTasks.py]: I have taken this suggestion and renamed the AMxWithHistTask to AMxTask.  This means that `extras` are carried around with every `AMx` measurement.

**Comment by SimonKrughoff** [comment]
I have also changed from a mode to a median for the aggregated computation.  The median is computed on the binned data, so only an estimate of the median is possible.  I chose a linear interpolation in the bin containing the median as the estimate of the median.

**Pull request merged**